stan/math/prim/prob/von_mises_cdf.hpp

```diff
@@ -0,0 +1,124 @@
+#ifndef STAN_MATH_PRIM_PROB_VON_MISES_CDF_HPP
+#define STAN_MATH_PRIM_PROB_VON_MISES_CDF_HPP
+
+#include <stan/math/prim/fun/constants.hpp>
+#include <stan/math/prim/prob/normal_cdf.hpp>
+#include <stan/math/prim/fun/modified_bessel_first_kind.hpp>
+#include <cmath>
+
+namespace stan {
+namespace math {
+
+namespace internal {
+
+template <typename T_x, typename T_k, typename T_p>
+return_type_t<T_x, T_k, T_p> von_mises_cdf_series(const T_x& x, const T_k& k,
+                                                  const T_p& p) {
+  using stan::math::cos;
+  using stan::math::sin;
+
+  double pi = stan::math::pi();
+  auto s = sin(x);
+  auto c = cos(x);
+  auto sn = sin(p * x);
+  auto cn = cos(p * x);
+
+  return_type_t<T_x, T_k, T_p> R = 0;
+  return_type_t<T_x, T_k, T_p> V = 0;
+
+  int n;
+  for (n = p - 1; n > 0; n--) {
+    auto sn_tmp = sn * c - cn * s;
+    cn = cn * c + sn * s;
+    sn = sn_tmp;
+    R = 1 / (2.0 * n / k + R);
+    V = R * (sn / n + V);
+  }
+  return 0.5 + x / (2 * pi) + V / pi;
+}
+
+template <typename T_x, typename T_k>
+return_type_t<T_x, T_k> von_mises_cdf_normalapprox(const T_x& x, const T_k& k) {
+  using std::exp;
+  using std::sqrt;
+  double pi = stan::math::pi();
+
+  auto b = sqrt(2 / pi) * exp(k) / modified_bessel_first_kind(0, k);
+  auto z = b * sin(x / 2);
+  double mu = 0;
+  double sigma = 1;
+
+  return normal_cdf(z, mu, sigma);
+}
+
+template <typename T_x, typename T_k>
+return_type_t<T_x, T_k> von_mises_cdf_centered(const T_x& x, const T_k& k) {
+  // if the scale is sufficiently small, approximate the cdf with a
+  // normal distribution, otherwise, use the expansion from scipy
+  double ck = 50;
+  double a[4] = {28, 0.5, 100, 5};
+  return_type_t<T_x, T_k> f;
+  if (k < ck) {
+    int p = value_of_rec(a[0] + a[1] * k - a[2] / (k + a[3]) + 1);
+    f = von_mises_cdf_series(x, k, p);
+    if (f < 0)
+      f = 0;
+    if (f > 1)
+      f = 1;
+  } else {
+    f = von_mises_cdf_normalapprox(x, k);
+  }
+  return f;
+}
```

Comment:
How about just hardcoding the stuff above?
```suggestion
using return_t = return_type_t<T_x, T_k>;
return_t f; // Should f be initialized to some value?
if (k < 50.0) {
  const auto p = value_of_rec(28.0 + 0.5 * k - 100.0 / (k + 5.0) + 1.0);
  f = von_mises_cdf_series(x, k, p);
  if (f < 0) {
    f = 0.0
    return f;
  } else if (f > 1.0) {
    f = 1.0;
    return f;
  }
} else {
  f = von_mises_cdf_normalapprox(x, k);
  return f;
}
return f;
}
```